lib/DataFileUtil/DataFileUtilImpl.py

```diff
@@ -471,6 +472,80 @@ def file_to_shock(self, ctx, params):
+           loading to Shock. This argument will be appended with the '.zip'
+           file extension prior to writing. If it is a directory, file name
+           of the created archive will be set to the directory name followed
+           by '.zip', possibly overwriting an existing file. Attempting to
+           pack the root directory is an error. ws_ref - list of references
+           to workspace objects which will be used to produce info-files in
+           JSON format containing workspace metadata and provenane structures
+           each. It produces new files in folder pointed by file_path (or
+           folder containing file pointed by file_path if it's not folder).
+           Optional parameters: attributes - user-specified attributes to
+           save to the Shock node along with the file.) -> structure:
+           parameter "file_path" of String, parameter "attributes" of mapping
+           from String to unspecified object, parameter "ws_refs" of list of
+           String
+        :returns: instance of type "PackageForDownloadOutput" (Output of the
+           package_for_download function. shock_id - the ID of the new Shock
+           node. node_file_name - the name of the file stored in Shock. size
+           - the size of the file stored in shock.) -> structure: parameter
+           "shock_id" of String, parameter "node_file_name" of String,
+           parameter "size" of String
+        """
+        # ctx is the context object
+        # return variables are: returnVal
+        #BEGIN package_for_download
+        if ctx['token'] is None:
+            raise ValueError('Authentication token required!')
+        file_path = params.get('file_path')
+        if not file_path:
+            raise ValueError('No file/directory provided.')
+        ws_refs = params.get('ws_refs')
+        if not ws_refs:
+            raise ValueError('No workspace references provided.')
+        dir_path = file_path
+        if not os.path.isdir(dir_path):
+            dir_path, temp_file_name = os.path.split(dir_path)
+        if not dir_path:
+            dir_path = '.'
+        dir_path = os.path.abspath(os.path.expanduser(dir_path))
+        objects = [{'ref': x} for x in ws_refs]
+        ws = Workspace(self.ws_url, token=ctx['token'])
+        items = ws.get_objects2({'no_data': 1, 'ignoreErrors': 1,
+                                 'objects': objects})['data']
+        for item in items:
+            item_info = item['info']
+            info_to_save = {'metadata': [item_info],
+                            'provenance': item['provenance']}
+            ws_name = item_info[7]
+            obj_name = item_info[1]
+            obj_ver = item_info[4]
+            info_file_name = 'KBase_object_details_' + ws_name + '_' + \
+                             obj_name + '_v' + str(obj_ver) + '.json'
+            info_file_path = os.path.join(dir_path, info_file_name)
+            with open(info_file_path, 'w') as info_file_writer:
```

Comment:
Since the data could contain unicode, should use io.open and specify an encoding.